### PR TITLE
DHFPROD-5608: Placement of new card in Load

### PR DESCRIPTION
**Pull request by briantang**
### Description

- Load cards now ordered by last updated.
- Pointed for develop as story is marked for 5.3

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by wooldridge** [comment]
@bsrikan, is there anywhere in the e2e tests that we test adding a new Load? That would be a good place to check that the added Load step is placed in the correct location.

**Comment by bsrikan** [comment]
> @bsrikan, is there anywhere in the e2e tests that we test adding a new Load? That would be a good place to check that the added Load step is placed in the correct location.

Yes we do. defaultIngestion.spec covers that

**Comment by wooldridge** [comment]
> > @bsrikan, is there anywhere in the e2e tests that we test adding a new Load? That would be a good place to check that the added Load step is placed in the correct location.
> 
> Yes we do. defaultIngestion.spec covers that

Would a test for this go here?

https://github.com/saarimrahman/marklogic-data-hub/blob/DHFPROD-5896/marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx#L168

@bsrikan or should be keep this testing in the RTL unit tests?

**Comment by briantang** [comment]
> > > @bsrikan, is there anywhere in the e2e tests that we test adding a new Load? That would be a good place to check that the added Load step is placed in the correct location.
> > 
> > 
> > Yes we do. defaultIngestion.spec covers that
> 
> Would a test for this go here?
> 
> https://github.com/saarimrahman/marklogic-data-hub/blob/DHFPROD-5896/marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx#L168
> 
> @bsrikan or should be keep this testing in the RTL unit tests?

My thinking was that sorting/ordering was more of a unit test to be handled in RTL rather than cypress. Please let us know your thoughts @bsrikan

File: marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx [hunk 1]
```diff
@@ -67,6 +67,14 @@ describe('Load Card component', () => {
     //Click on 'Yes' button
     fireEvent.click(getByTestId('testLoadXML-to-FlowA-Confirm'));
 
+    //verify Load cards are ordered by Last Updated
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: Is this the right place to add these assertions. Previous step is clicking on Yes to add to existing flow and redirected to run tile, so this is like breaking the chain of events. This scenario probably deserves to be a separate test. You could add it to this one before load step is added to a flow, but then you also would have to update the test description.

Comment (briantang) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: Okay, I will work on refactoring the test here in the same file then. Thanks!

Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: > Is this the right place to add these assertions. Previous step is clicking on Yes to add to existing flow and redirected to run tile, so this is like breaking the chain of events. This scenario probably deserves to be a separate test. You could add it to this one before load step is added to a flow, but then you also would have to update the test description.

@bsrikan but generally are we doing roundtrip data CRUD in the unit tests? Seems like when we test POSTs, we're just testing that the POST gets called and not going about checking the state of the existing payload for that view. I could be wrong. Is there an example similar to this? I will look...

Comment (bsrikan) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: You can test for something like this:

```
//before update
    expect(loadCards[0]).toHaveTextContent('Add New');
    expect(loadCards[1]).toHaveTextContent('testLoadXML');
    expect(loadCards[2]).toHaveTextContent('testLoad123');
    expect(loadCards[3]).toHaveTextContent('testLoad');

//update timestamp for testLoadXML
data.loadData.data[1].lastUpdated = "1990-04-15T14:22:54.057519-07:00"

//after update
    expect(loadCards[0]).toHaveTextContent('Add New');
    expect(loadCards[2]).toHaveTextContent('testLoad123');
    expect(loadCards[3]).toHaveTextContent('testLoad');
    expect(loadCards[1]).toHaveTextContent('testLoadXML');
```

Comment (briantang) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: good idea, thanks. Will work on that.

Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: @bsrikan, nice. We can avoid mucking with the CRUD and just change the source data and see that the UI adjusts.

Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: @briantang, I like @bsrikan's suggestion of making a separate test.

Comment (briantang) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: Noted, will do. Thanks all :)

Comment (bsrikan) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:70]: @wooldridge you are right about the roundtrip. For RTL unit tests, our mock data is setting up the response for POST request. So its like testing with the assumption that server responds as expected. Now its upto the UI to render that correctly. The same is tested with scenario above. We check for the order of load cards as is. Then update the timestamp, which is to mock the edit/save load configuration, when the timestamp is updated and verify the order of load cards again.
The only roundtrip from server for this ticket is the timestamp update on edit/save. But is it worth a Cypress test, probably not IMO. If a scenario comes up in the future where we edit the settings or step details, we could additionally test for placement based on last updated.

**Comment by briantang** [comment]
RTL test modification is pushed, ready for re-review.

**Review by bsrikan — APPROVED** [review]

File: marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx (hunk 1 above)
Comment (bsrikan) [reply marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx]: @wooldridge we were not able to test as suggested; rerender was not calling useEffect and so we were not able to verify the sort order after timestamp was updated within the test. Instead, the mock data itself has unordered list, so we verify that it renders the cards in the expected order.

**Review by Sanjeevani19 — APPROVED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]

**Pull request merged**